--- /usr/lib/enigma2/python/Components/Converter.org/ServiceName.py
+++ /usr/lib/enigma2/python/Components/Converter/ServiceName.py
@@ -42,7 +42,16 @@
 		elif self.type == self.PROVIDER:
 			return self.getServiceInfoValue(info, iServiceInformation.sProvider, ref)
 		elif self.type == self.REFERENCE:
-			return self.getServiceInfoValue(info, iServiceInformation.sServiceref, ref)
+			#return self.getServiceInfoValue(info, iServiceInformation.sServiceref, ref)
+			serv_ref = self.getServiceInfoValue(info, iServiceInformation.sServiceref, ref)
+			pos = -1
+			if ref:
+				pos = ref.toString().rfind(':http')
+			if (serv_ref == 'N/A' or pos != -1) and ref:
+				serv_ref = ref.toString().replace('%3a', ':')
+			return serv_ref
+		else:
+			return
 
 	text = property(getText)
 
